src/vs/code/electron-main/window.ts

```diff
@@ -167,9 +167,10 @@ export class CodeWindow implements ICodeWindow {
 		}
 
 		let useCustomTitleStyle = false;
-		if (isMacintosh && (!windowConfig || !windowConfig.titleBarStyle || windowConfig.titleBarStyle === 'custom')) {
+		if ((!isMacintosh && windowConfig && windowConfig.titleBarStyle === 'custom') ||
```

Comment:
@sbatten very hard to read this check, maybe extract some booleans from the windowConfig title style to make this better?